### PR TITLE
Tta #33 search bar fix server

### PR DESCRIPTION
**Pull request by mnmin**

### CODE REVIEW

File: src/controllers/user.js
Comment (vherus) [inline src/controllers/user.js:63]: Let's get rid of this

**Review by JohnXander — APPROVED** [review]

File: src/domain/user.js
Comment (vherus) [inline src/domain/user.js:177]: The purpose of this function is to provide a simple interface to find users by their first name. This change is making the function more general, which is what `_findMany` already does. I don't think this is a useful change to make.

If you want to add functionality to be able to search users by multiple fields, you should create new things for that and use the existing functionality where you're able to

**Review by vherus — CHANGES_REQUESTED** [review]

File: src/controllers/user.js
Comment (mnmin) [reply src/controllers/user.js]: removed

File: src/domain/user.js
Comment (mnmin) [reply src/domain/user.js]: Changed the functionality of the functions findManyByFirstName and getAllByFirstName

File: src/domain/user.js
Comment (vherus) [inline src/domain/user.js:219]: Since there are no unit tests in this app, please manually test every route that relies on this function before merging the PR.

**Review by vherus — APPROVED** [review]
Approved, but don't merge until you've completed the testing task in case it breaks something (the client will start failing if anything is broken so it's important)

File: src/domain/user.js
Comment (mnmin) [reply src/domain/user.js:219]: manually tested every routes and one isn't working. Currently working on it

**Pull request merged**
